[Tizen Studio] 4.6 Release Notes (#1595)
Samsung/tizen-docs: docs/application/tizen-studio/release-notes/4-6-release-notes.md

# Tizen Studio 4.6 Release Notes

- Release Date: Mar 31, 2022

## IDE and Tools

### New Feature

- macos Monterey support
  - Tizen IDE is now supported on macos Monterey
  - Visual Studio Code Extension for Tizen is now supported on macos Monterey
- Multi App and Hybrid App support
  - Added the following Multi app and Hybrid App support in CLI(TZ), VS and VSCode extension for Tizen

>   **Multi App**

Author: Done

>   **Multi App**

Reviewer: Line 15: I noticed that this part is indented in comparison with lines 29, 42, and 50. So by making it un-indented, it will be in alignment with the other lines. Please address this issue. 
```suggestion
**Multi App**
```

| Widget   App(Main)     | 0                  | 0                       | 0                      | 0                         | M                      | M                      | 0                     |
| Component   App(Main)  | 0                  | 0                       | 0                      | 0                         | M                      | M                      | 0                     |
| Shared   Library(Main) | 0                  | 0                       | 0                      | 0                         | M                      | M                      | 0                     |
| Static   Library(Main) | 0                  | 0                       | 0                      | 0                         | M                      | M                      | 0                     |
| Watch   App(Main)      | 0                  | M                       | 0                      | 0                         | M                      | M                      | 0                     |

> **Web Apps**

Author: Done

> **Web Apps**

Reviewer: @adityaaswani I can see that you have added space but it still appears that there is no space between the table and line 42 ("Web Apps") on the staging area. What can be done about this to fix it?

| Apps               | Native Service App (Dependent) | Native Widget App (Dependent) | Web App (Dependent) | Web Widget (Dependent) | Dotnet UI app (Dependent) | Dotnet Service App (Dependent) |   |
|--------------------|--------------------------------|-------------------------------|---------------------|------------------------|---------------------------|--------------------------------|---|
| Web   App(Main)    | M                              | M                             | 1                   | M                      | M                         | M                              |   |
| Web   Widget(Main) | 0                              | 0                             | 0                   | 0                      | 0                         | 0                              |   |

>   **Hybrid App**

Author: Done

>   **Hybrid App**

Reviewer: @adityaaswani I can see that you have added space but it still appears that there is no space between the table and line 50 ("Hybrid App") on the staging area. What can be done about this to fix it?

  Developers can now have multiple dependent applications **of different types** in a single workspace and perform all App life cycle events like app creation, building, packaging, installing and testing.
    
| Main App          | Possible Sub App Combinations |
|-------------------|-------------------------------|
| Native   Main App | Native + Dotnet               |
| Dotnet   Main App | Dotnet + Native               |
| Web   Main App    | Web + Native +   Dotnet       |

- CLI

>   - Added support for importing Hybrid and Multi projects

Reviewer: To maintain consistency with other bullet points, the sentence should end with a period. 
```suggestion
  - Added support for importing Hybrid and Multi projects.
```

  - Fixed the failure to open yaml config files for native apps in VS extension for Tizen. 
  - Fixed the issue due to which there was no option to import Tizen TV web app projects in VS & VSCode extension for Tizen.
  - Fixed the issue in Import functionality and added missing support to import wgt(Web Application Package file) in VS & VSCode extension for Tizen

### Deprecated
  - Bridge Networking support in emulators will be deprecated in future Tizen Studio releases.

### Known Issues

- Installer
  - In macOS, Tizen Studio related file are marked as Malware, hence it is recommended to select **Anywhere** under allow apps as described in https://www.imore.com/how-open-apps-anywhere-macos-catalina-and-mojave
- Tizen Studio
  - Valgrind profiling is not supported in Tizen 6.5 platform
  - In macOS, if the UI perspectives are not displayed properly after updating Tizen Studio, it is recommended to restart the IDE. This issue is because of the uncleared cache from the previous installation.
  - In case if you see any old perspective, it is recommended to create a new workspace and import the projects to the new workspace to resolve the issue.

>   - In macOS from catalina and above versions, Native Templates-5.5 will not build with CLI when compiler is set to **gcc**.

Reviewer: Since “Catalina” is a proper noun specifying a macOS version, it should be capitalized. 
I am not sure why the words “Native Templates” have been capitalized here. Can you please confirm this? 
An article “the” is missing before the word “compiler”. 

```suggestion
  - In macOS from Catalina and above versions, Native Templates-5.5 will not build with CLI when the compiler is set to **gcc**.
```

- Common

>   - If you install Tizen Studio in a directory that requires administrator privileges for access. For example, for **C:\ProgramFiles** location administrator privilege is needed to run the Tizen SDK tools. The Tizen Installer and the baseline SDK Installer alerts you if you try to install it into such a directory.

Reviewer: The first line seems incomplete, so I have made suggestions to make it understandable. Please let me know if this delivers the meaning properly. 
“The” has been removed before “Tizen SDK tools”, “Tizen Installer”, "Baseline SDK Installer" since it is recommended to not use “the” before nouns such as tool names. 

```suggestion
  - If you install Tizen Studio in a directory that requires administrator privileges for access, an alert will be given. . For example, for **C:\ProgramFiles** location administrator privilege is needed to run Tizen SDK tools. Tizen Installer and Baseline SDK Installer alerts you if you try to install it into such a directory.
```

- Web IDE
  - The preview tab in the Web Page Editor sometimes does not appear properly. Use an alternative feature, named Web SDK HTML Editor, which has enhanced features compared to the Web Page Editor. Instead of the preview tab in the Web Page Editor, use the preview feature **Ctrl + 4** of the Web SDK HTML Editor.
  - In Rapid Development Support (RDS) mode, the web unit test result is not updated.
  - Remote Inspector does not allow audit in the updated version of Chrome.
- Certificate Manager
  - Overwriting a duplicate certificate profile in the migration wizard works incorrectly on macOS.
- Emulator
  - In the Tizen 5.5 version, the emulator images app un-installation might take some time to complete if the app name contains more than 14 characters. A patch for the same is expected to be released in the next update.
  - To use Tizen Emulator, use Intel VTx supported CPU and the latest version of the graphic card driver provided by the vendor. To verify the prerequisites for Tizen Emulator, see [Prerequisites for Tizen Studio](../setup/prerequisites.md).

>     - If the host machine is using NVIDIA&reg; Optimus&reg; technology on either Ubuntu or Windows, you must set the Tizen Emulator to run with your NVIDIA&reg; graphics card. For Ubuntu, verify the [bumblebee project](https://wiki.ubuntu.com/Bumblebee). For Windows, select **High Speed NVIDIA&reg; Processor** as **Preferred Graphics Processor** in the NVIDIA&reg; control panel.

Reviewer: “The” has been removed before “Tizen Emulator” since it is recommended to not use “the” before nouns such as tool names. Also, "NVIDIA Control Panel" is a proper noun, so it should be capitalized. 
```suggestion
    - If the host machine is using NVIDIA&reg; Optimus&reg; technology on either Ubuntu or Windows, you must set Tizen Emulator to run with your NVIDIA&reg; graphics card. For Ubuntu, verify the [bumblebee project](https://wiki.ubuntu.com/Bumblebee). For Windows, select **High Speed NVIDIA&reg; Processor** as **Preferred Graphics Processor** in the NVIDIA&reg; Control Panel.
```

>     - On Ubuntu, if the graphics driver is out-of-date, your Ubuntu desktop session occasionally logs out while launching Emulator Manager, or the emulator skin is displayed improperly. Verify the prerequisites and upgrade to the latest graphics driver.

Reviewer: I have restructured the sentence for better readability. Please let me know if this looks okay to you. 
```suggestion
    - On Ubuntu, if the graphics driver is out-of-date, while launching Emulator Manager, your Ubuntu desktop session occasionally logs out or the emulator skin is displayed improperly. Verify the prerequisites and upgrade to the latest graphics driver.
```

  - On Windows, depending on your OS theme (such as Non-Aero themes and Windows XP themes), a display surface can be erased for a while if the emulator window is covered with another window. If you click the emulator window, the display surface runs correctly again.

>   - On Windows, if an error with message "failed to allocate memory" occurs while executing the emulator, try the following:

Reviewer: An article “the” is missing before the word “message”. 
```suggestion
  - On Windows, if an error with the message "failed to allocate memory" occurs while executing the emulator, try the following:
```

    - Close some other programs and try to launch the emulator again.
    - If the RAM size is set to 768 or 1024 MB for the VM in Emulator Manager, change it to 512 MB.
    - Increase the user area of the virtual memory in the system to 3 GB by entering the **bcdedit /setincreaseuserva 3072** command on the console with administrator rights (only in Windows 7), and reboot.
  - If you use a MacBook Pro that has both Intel HD and NVIDIA&reg; GPUs, the emulator can terminate unexpectedly when you execute the emulator with OpenGL ES version 1.1 or 2.0. Verify the emulator configuration in Emulator Manager and on the **General** tab in the emulator configuration window, set OpenGL ES version to version 2.0, or to version 3.0.

>   - When you launch Emulator Manager in the Tizen IDE, the shortcut image of Emulator Manager may not be displayed properly.

Reviewer: It is recommended to not use "the" before tool names.  
```suggestion
  - When you launch Emulator Manager in Tizen IDE, the shortcut image of Emulator Manager may not be displayed properly.
```

>   - Basic Web applications are not installed on SD cards.

Reviewer: Can you please tell why the word “web” has been capitalized here?
```suggestion
  - Basic web applications are not installed on SD cards.
```

  - To use Tizen Emulator in Tizen platform 3.0 or lower, disable the CPU VT option in the **HW Support** tab of Emulator Configuration.
- CLI and SDB
  - Tizen Studio does not support the [Smart Development Bridge](../common-tools/smart-development-bridge.md) (SDB) bash auto-completion on Windows (it is available on Ubuntu and macOS).
- Dynamic Analyzer
  - When analyzing applications on commercial devices running Tizen 3.0, newly-released or after a firmware update, the following problems exist:

>     - The Core Frequency information is not shown.

Reviewer: I am not sure why the words “core frequency” have been capitalized here.
```suggestion
    - The core frequency information is not shown.
```

    - The screenshots on the scene transitions feature will not work.

>   - When analyzing applications on the Tizen 4.0 emulator or reference device, the startup profiling information is not shown.

Reviewer: For noun such as tool names, it is recommended not to use “the”. 
```suggestion
  - When analyzing applications on Tizen 4.0 Emulator or reference device, the startup profiling information is not shown.
```

  - The UI Hierarchy viewer feature and startup profiling are not performed simultaneously.

>   - The Dynamic Analyzer cannot show lifecycle information for Web applications.

Reviewer: Can you please tell why the word “web” has been capitalized here?
```suggestion
  - The Dynamic Analyzer cannot show lifecycle information for web applications.
```

  - Widget applications cannot be profiled with the Dynamic Analyzer. They are hidden in the application list on the toolbar for all Tizen platforms, except Tizen 2.3.2.
  - The Dynamic Analyzer sometimes gets stuck which is caused by an internal security checking program.
- Visual Studio

>   - NUI XAML application build error can be resolved by doing clean and then build again.

Reviewer: Line 123: Should the word “clean” and “build” stay as it is considering it specifies the feature itself? Or should we restructure the sentence?

- VS Mac

>   - NUI XAML application build error can be resolved by doing clean and then build again.

Reviewer: Line 125: Should the word “clean” and “build” stay as it is considering it specifies the feature itself? Or should we restructure the sentence?

Author: We need both clean and build as only build will resut in an error

# Tizen Studio 4.6 Release Notes

- Release Date: Mar 31, 2022

## IDE and tools

### New feature

- macOS Monterey support
  - Tizen IDE is now supported on macOS Monterey
  - Visual Studio Code Extension for Tizen is now supported on macOS Monterey
- Multi App and Hybrid App support

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Author: The links are not yet live but under development

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Reviewer: @adityaaswani thank you for your prompt reply. Can you also please change the hyperlinks into relative paths? For example: (https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/) this should be changed to (../tizen-core/tizen-core-cli).

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Author: Done

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Reviewer: Thank you @adityaaswani for the rapid change. Once the other two links are developed, please update those hyperlinks to relative paths as well.

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Reviewer 2: Hello.

1. When relative path are used, the file name is used.
   So it should be  (../tizen-core/tizen-core-cli.md). Same to other hybrid pages. Please refer file name when you insert hyper links. If the link is valid, you'd be possible to check on git diff, too.

2. The hybrid pages are working on https://github.com/Samsung/tizen-docs/pull/1608 and https://github.com/Samsung/tizen-docs/pull/1612
   I think when PR 1608 and PR 1612 are merged, the link will work.

3. The release date is changed to 03.30 in 4-6-release-notes.md.
   Please also update  the release date 03.31 in toc_all.md, too.

>   - Added the following Multi app and Hybrid app support in [CLI (TZ)](https://docs.tizen.org/application/tizen-studio/tizen-core/tizen-core-cli/), [VS](https://docs.tizen.org/application/vstools/Tizen/hybrid), and [VSCode](https://docs.tizen.org/application/vscode-ext/Tizen/hybrid) extension for Tizen

Author: Done the changes

| Watch   App (Main)      | 0                  | M                       | 0                      | 0                         | M                      | M                      | 0                     |

**Web Apps**

| Apps               | Native Service App (Dependent) | Native Widget App (Dependent) | Web App (Dependent) | Web Widget (Dependent) | Dotnet UI app (Dependent) | Dotnet Service App (Dependent) |   |
|--------------------|--------------------------------|-------------------------------|---------------------|------------------------|---------------------------|--------------------------------|---|
| Web   App (Main)    | M                              | M                             | 1                   | M                      | M                         | M                              |   |
| Web   Widget (Main) | 0                              | 0                             | 0                   | 0                      | 0                         | 0                              |   |

  **Hybrid App**
  

>   Developers can now have multiple dependent applications **of different types** in a single workspace and perform all app life cycle events like app creation, building, packaging, installing, and testing.

Reviewer: I think the words "of different types" shouldn't be made bold here. 

```suggestion
  Developers can now have multiple dependent applications of different types in a single workspace and perform all app life cycle events like app creation, building, packaging, installing, and testing.
```
